test/server/cards/02_SHD/units/ReputableHunter.spec.ts

```diff
@@ -0,0 +1,63 @@
+describe('Reputable Hunter', function() {
+    integration(function(contextRef) {
+        describe('Reputable Hunter\'s decrease cost ability', function() {
+            it('should cost 3 if there are no bounties on enemy units', () => {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['reputable-hunter'],
+                        base: 'energy-conversion-lab',
+                    }
+                });
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.reputableHunter);
+
+                expect(context.player1.countExhaustedResources()).toBe(3);
+            });
+
+
+            it('should cost 2 after we play a bounty on a unit', () => {
```

Comment:
Just a note - for tests of similar abilities in the future we won't need to check that dynamically gained bounties work correctly, we can assume it works. That system is relatively new so the first few units we added had an explicit check for it just to confirm that it works, and it's also fine to include in this one as well. (Not great separation of test concerns but we do what we can)